tests/test_models.py

```diff
@@ -1,22 +1,188 @@
 # -*- coding: utf-8 -*-
+from django.conf import settings
+from django.core.exceptions import ValidationError
 from django.test import TestCase
 
-from djangocms_googlemap.models import GoogleMap
+from djangocms_googlemap.models import (
+    GoogleMap, GoogleMapMarker, GoogleMapRoute, get_templates,
+    ZOOM_LEVEL_CHOICES, MAP_TYPE_CHOICES, TRAVEL_MODE_CHOICES
+)
 
+from .helpers import get_filer_image
 
-class GoogleMapTestCase(TestCase):
+
+SAMPLE_STYLES = """
+[{"elementType": "geometry", "stylers": [{"color": "#242f3e"}]},
+{"elementType": "labels.text.stroke", "stylers": [{"color": "#242f3e"}]},
+{"elementType": "labels.text.fill", "stylers": [{"color": "#746855"}]}]
+"""
+
+
+class GoogleMapModelTestCase(TestCase):
 
     def setUp(self):
+        self.icon = get_filer_image("pin.png")
+
+    def tearDown(self):
+        if self.icon:
+            self.icon.delete()
+            del self.icon
+            with self.assertRaises(AttributeError):
+                print(self.icon)
+        GoogleMap.objects.filter(pk=1).delete()
+        self.assertEqual(GoogleMap.objects.all().count(), 0)
+        GoogleMapMarker.objects.filter(pk=1).delete()
+        self.assertEqual(GoogleMapMarker.objects.all().count(), 0)
+        GoogleMapRoute.objects.filter(pk=1).delete()
+        self.assertEqual(GoogleMapRoute.objects.all().count(), 0)
```

Comment:
These aren't really necessary, are they? The database gets cleaned up after each test automatically…

Reply:
Is that always true? Was pointed by @kinkerl to make sure to clean that stuff up

Reply:
Yeah, it's true as long as the items are only in database, because the database gets rolled back to its state before the test. If there are any files created in the filesystem (such as the temporary directories), those would have to be cleaned up explicitly.

This is true for the regular `TestCase`, which simply wraps the entire test in a transaction which it aborts at the end; it's slightly different if you were using `TransactionTestCase`, but even then, the database would get cleaned up, just in a different way.